Lib/subprocess.py

```diff
@@ -1307,15 +1307,18 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
                     exception_name, hex_errno, err_msg = (
                             errpipe_data.split(b':', 2))
-                except ValueError:
+                    # The encoding here should match the encoding
+                    # written in by the subprocess implementations
+                    # like _posixsubprocess
+                    err_msg = err_msg.decode()
+                except (ValueError, UnicodeError):
                     exception_name = b'SubprocessError'
                     hex_errno = b'0'
-                    err_msg = (b'Bad exception data from child: ' +
-                               repr(errpipe_data))
+                    err_msg = "Bad exception data from child: {!r}".format(
```

Comment:
You could use f-string here and fit the expression in one line.

In any case please use single quotes. They are default quotes for string. I suppose you used double quotes because the string contained single quotes around %s. But now they are gone.